Unity/Assets/Plugins/SignalR/SignalRInterface.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Text.Json;
+using System.Linq;
+using UnityEngine;
+
+public class SignalRInterface : SignalRLib
+{
+    public ConcurrentDictionary<string, Action<object[]>> Functions = new ConcurrentDictionary<string, Action<object[]>>();
+    public bool IsConnected { get; set; } = false;
+
+    public SignalRInterface()
+    {
+        HandlerInvoked += SignalRInterface_HandlerInvoked;
+        ConnectionStarted += SignalRInterface_ConnectionStarted;
+    }
+
+    private void SignalRInterface_ConnectionStarted(object sender, ConnectionEventArgs e)
+    {
+        Debug.Log(e.ConnectionId);
+        IsConnected = true;
+    }
+
+    private void SignalRInterface_HandlerInvoked(object sender, HandlerEventArgs e)
+    {
+        try 
+        { 
+            Action<object[]> action;
+
+            if (Functions.TryGetValue(e.HandlerName, out action))
+            {
+                var arrayParameters = JsonSerializer.Deserialize<List<object>>(e.Payload);
```

Comment:
Should first validate that the payload is actually Json (as we want to allow plain Strings).

Reply:
Im working on it... Im so sorry about be late, but these days Im a little bit busy...